packages/search/src/rules/legacyActionRule.ts

```diff
@@ -26,11 +29,28 @@ const isLegacyAction = (model: ActionModel) => {
   switch (model.type) {
     case 'Custom':
     case undefined:
-      return legacyCustomActions.some(
-        (action) => action.name === model.name && action.type === model?.type,
-      )
+      // Legacy action has no version, while newer ones have a version 2+
+      return !model.version && LEGACY_CUSTOM_ACTIONS.has(model.name)
   }
   return false
 }
 
-const legacyCustomActions = [{ type: undefined, name: 'If' }]
+const LEGACY_CUSTOM_ACTIONS = new Set([
+  'If',
+  'PreventDefault',
+  'StopPropagation',
+  'Copy To Clipboard',
+  'CopyToClipboard',
+  'UpdateVariable',
+  'Update Variable',
+  'Update URL parameter',
+  'updateUrlParameters',
+  'UpdateQueryParam',
+  'Update Query',
+  'Fetch',
+  'SetTimeout',
+  'SetInterval',
+  'FocusElement',
+  'Debug',
+  'GoToURL',
```

Comment:
I believe the `TriggerEvent` action should also be considered a legacy action?
```suggestion
  'GoToURL',
  'TriggerEvent',
```

Reply:
Very true! I committed your suggestion. I tried to be thorough, but might have missed something still.